src/net.rs

```diff
@@ -236,6 +240,9 @@ impl NetworkStream for HttpStream {
             Https(ref mut inner) => inner.get_mut().peer_name()
         }
     }
+
+    fn mark_dead(&mut self) {
```

Comment:
No, the "built-in" `HttpStream` is created and then dropped when used by the client currently, so it doesn't particularly care about the state of the response header; it's going to close anyhow.

This is for custom `NetworkStream`s that are part of a pool and have additional fields that track health.